core-bundle/src/Controller/SitemapController.php

```diff
@@ -41,7 +41,12 @@ public function __invoke(Request $request): Response
         $rootPages = $pageModel->findPublishedRootPages(['dns' => $request->server->get('HTTP_HOST')]);
 
         if (null === $rootPages) {
-            return new Response('', Response::HTTP_NOT_FOUND);
+            // We did not find root pages by matching host name, let's fetch those that do not have any domain configured
+            $rootPages = $pageModel->findPublishedRootPages(['dns' => '']);
```

Comment:
FYI: At first, I wanted to combine it in one query but that wouldn't work correctly. If you had one root page with a domain configured and one not, the one with the domain configured would then also include the ones from the one without domain configured which is not what we want.